src/workflow_components/workflow.py

```diff
@@ -170,40 +171,29 @@ def __init__(self, name: Optional[str], path: str):
     def from_dict(obj_dict: Dict[str, Any]) -> "Workflow":
         w = Workflow(name=obj_dict.get("name"), path=obj_dict["path"])
 
-        trigger = []
-        if isinstance(obj_dict["on"], str):
-            trigger = [obj_dict["on"]]
-        elif isinstance(obj_dict["on"], list):
-            trigger = []
-            for elem in obj_dict["on"]:
-                if isinstance(elem, dict):
-                    trigger.extend(elem.keys())
+        w.trigger = parse_workflow_trigger(obj_dict["on"])
+
+        # Handling special case of workflow_run
+        # When we meet it, we want to create a special relation from the triggering workflow,
+        # to the triggered one.
+        # There are cases where the triggering workflow wasn't loaded yet.
+        # In that case we creating a stub node for it,
+        # and once we'll meet it, we'll enrich it.
+        if "workflow_run" in w.trigger:
```

Comment:
Can this be in paring function as well?

Reply:
I thought deeply about it.
Once we find the workflow run trigger, we need to find the workflow and update the relations
So the parsing function will be either 3 lines (which is useless), or receive many parameters (which will be harder to test).
I prefer to leave it as is